internal/data/data_manager.go

```diff
@@ -539,3 +540,44 @@ func (dm *dataManager) WriteNewMessage(ctx context.Context, newMsg *NewMessage)
 func (dm *dataManager) WaitStop() {
 	dm.messageWriter.close()
 }
+
+func (dm *dataManager) DeleteData(ctx context.Context, dataID string) error {
+	id, err := fftypes.ParseUUID(ctx, dataID)
+	if err != nil {
+		return err
+	}
+
+	data, err := dm.database.GetDataByID(ctx, dm.namespace.Name, id, false)
+	if err != nil {
+		return err
+	}
+	if data == nil {
+		return i18n.NewError(ctx, coremsgs.Msg404NoResult)
+	}
+	if data.Blob != nil && data.Blob.Hash != nil {
+		fb := database.BlobQueryFactory.NewFilter(context.Background())
+		blobs, _, err := dm.database.GetBlobs(ctx, fb.And(fb.Eq("data_id", data.ID), fb.Eq("hash", data.Blob.Hash)))
```

Comment:
Are we intentionally querying across namespaces here? I would expect it to be scoped to the namespace managed by this instance of `dataManager`.